stub/include/neurala/image/dto/ImageMetadata.h

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (c) Neurala Inc. 2013-2021
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and
+ * associated documentation files (the "Software"), to deal in the Software without restriction,
+ * including without limitation the rights to use, copy, modify, merge, publish, distribute,
+ * sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:  The above copyright notice and this
+ * permission notice (including the next paragraph) shall be included in all copies or substantial
+ * portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT
+ * NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#ifndef NEURALA_IMAGE_DTO_IMAGE_METADATA_H
+#define NEURALA_IMAGE_DTO_IMAGE_METADATA_H
+
+#include <cstddef>
+#include <ostream>
+#include <string>
+#include <utility>
+
+namespace neurala
+{
+namespace dto
+{
+/**
+ * @brief Image metadata transfer class.
+ *
+ * Describes an image using dimensions, color space, data layout and data type.
+ */
+class ImageMetadata
+{
+public:
+	using size_type = std::size_t;
+
+private:
+	size_type m_width{};
+	size_type m_height{};
+	std::string m_colorSpace;
+	std::string m_layout;
+	std::string m_datatype;
+
+public:
+	ImageMetadata() = default;
+
+	/**
+	 * @brief Constructs a new @ref ImageMetadata with the provided arguments.
+	 *
+	 * @param width      image widht
+	 * @param height     image height
+	 * @param colorSpace image color space
+	 * @param layout     image data layout
+	 * @param datatype   image data type
+	 */
+	ImageMetadata(size_type width,
+	              size_type height,
+	              std::string&& colorSpace,
+	              std::string&& layout,
+	              std::string&& datatype) noexcept
```

Comment:
You're right. I have some reservations about by-value parameters because if the parameter is an rvalue, it adds an unnecessary move, but having to clone lvalue strings when you call is even worse.